service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -24,35 +25,85 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		return nil, microerror.Mask(err)
 	}
 
-	var configMap *corev1.ConfigMap
+	var ingressControllerReplicas int32
 	{
-		v := map[string]string{
-			"baseDomain":   key.TenantEndpoint(cr, cc.Status.Endpoint.Base),
-			"clusterDNSIP": r.dnsIP,
-			"clusterID":    key.ClusterID(&cr),
+		// We set the number of replicas to the number of worker nodes. This is
+		// set by the workercount resource which returns a map with each node
+		// pool and its current number of nodes.
+		for _, v := range cc.Status.Worker {
+			ingressControllerReplicas += v.Nodes
 		}
+	}
+
+	// We limit the number of replicas to 20 as running more than this does
+	// not make sense.
```

Comment:
@puja108 @corest This is now capped at 20 replicas based on the upstream recommendation. PTAL we can also reduce to 12 if we think that is better.